src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -64,7 +64,13 @@ internal sealed class RegexCache
         /// <summary>Gets or sets the maximum size of the cache.</summary>
         public static int MaxCacheSize
         {
-            get => s_maxCacheSize;
+            get
+            {
+                lock (SyncObj)
```

Comment:
Can you explain why locking a read to an int is necessary?

Reply:
Oh, I see the issue. Hmm, pretty hard to reason about whether we might overlook this possibility in other contexts.